pyro/contrib/minipyro.py

```diff
@@ -324,22 +313,19 @@ def elbo(model, guide, *args, **kwargs):
     # terms, this means our loss is constructed as an expectation w.r.t. the joint
     # distribution defined by the guide.
     model_trace = trace(replay(model, guide_trace)).get_trace(*args, **kwargs)
-    # We will accumulate the various terms of the ELBO in `elbo`.
+    # Return (-elbo) since by convention we do gradient descent on a loss and
+    # the ELBO is a lower bound that needs to be maximized. Given this,
+    # loop over all the sample sites in the model and add the corresponding
+    # - log p(z) term to the ELBO. Same thing for guide but adding + log p(z)
+    return elbo_terms(guide_trace) - elbo_terms(model_trace)
```

Comment:
Hmm, while this is fewer lines of code, it is more abstract. Again I think we need different [complexity metrics](https://en.wikipedia.org/wiki/Cyclomatic_complexity) for main Pyro and minipyro: In Pyro we should prefer well-factored code; and in minipyro we should [prefer monolithic blocks of code](http://number-none.com/blow/blog/programming/2014/09/26/carmack-on-inlined-code.html) and specifically blocks of code that look like minified versions of main Pyro.